### ISSUE TITLE
[global] implement --no-timeout option

### ISSUE THREAD

**Issue by bmr-cymru** [sosreport/sos#337]
We've supported timeouts for a while now, but what if something's timing out and the user wants to give it a little longer to run?

We could allow timeouts to be passed in on the command line (bit icky since plugins can control their own timeout setting via the command collection APIs).

Simplest seems to be to allow the command timeout support to be disabled by the user, e.g.:

```
# sosreport -v --batch --no-timeout
```

T/C/F?

**Comment by bmr-cymru** [comment]
Some musings on this issue from PR 971:

What I'm leaning toward now is something like:

```
# sosreport --timeout=none

# sosreport --timeout=10
```

And maybe later if we want to get fancy, allowing a syntax to set it on a per-plugin basis, e.g.:

```
# sosreport --timeout=general=300

# sosreport --timeout=general=300:10
```
The first notation would set the general plugin timeout to 300s, while leaving everything else at the default, and the 2nd would set general to 300s, and the default to 10s.

I'm not married to the syntax - if anyone has a less ugly suggestion I'd be happy to use it. The simple syntax should be straightforward enough. We can add per-plugin settings later if useful (even this isn't too tricky I don't think - just a case of picking an attribute for the per-plugin timer and wiring it up to calls that accept a timeout).

**Comment by bmr-cymru** [comment]
(so the form of `--timeout` is a colon-separated list of `plugname=timeout_seconds`, with an optional terminating `global_timeout_seconds` at the end).

**Comment by battlemidget** [comment]
I actually quite like the proposed syntax

**Comment by bmr-cymru** [comment]
OK, cool - I think this could be something we can go with. It'll also do an end-run on all those annoying buglets "plugin 'qux' times out every time I do 'xyzzf'" - users can directly work around it as it happens.

What I meant by ugly is the 'double equal' thing, but I can't think of anything better (although `--timeout general=300:apache=300:15` looks fine and would work with the same `OptionParser` stuff).

**Comment by ChrisCheney** [comment]
Something to note is that the no timeout / adjustable limit should also be applied to the compression time of the final tarball. Currently you can get all the way to the end and then crash due to the overall tarball taking longer than 300s to compress.

**Comment by pmoravec** [comment]
Compressing should be done without a timeout already. See #910 .

**Comment by bmr-cymru** [comment]
Anyway: we ran out of time on 3.6 for this, but we will reconsider it for 3.7.

**Comment by TurboTurtle** [comment]
I believe everything brought up in here is addressed by #1499?

`--plugin-timeout` for global control and `-k plugin.timeout` for specific control, with `0` being used in either case for a no timeout approach.

**Comment by bmr-cymru** [comment]
Yup, looks to be, thanks!
